### PR TITLE
Adding scrolling to current highlighted block.

### PR DESCRIPTION
**Pull request by moniika**

### CODE REVIEW

File: blockly-ws-search/workspace_search.js [hunk 1]
```diff
@@ -395,4 +401,75 @@ class WorkspaceSearch {
...
+    // Workspace scale, used to convert from workspace coordinates to pixels.
+    const scale = this.workspace_.scale;
+
+    // Block bounds in pixels relative to the workspace origin (0,0 is centre).
+    const top = blockTop * scale;
+    const bottom = blockBottom * scale;
+    const left = blockLeft * scale;
+    const right = blockRight * scale;
+    const width = blockWidth * scale;
+    const height = blockHeight * scale;
```
Comment (alschmiedt) [inline blockly-ws-search/workspace_search.js]: I would maybe consider combining these and then you could just have one set of constants and have bock in the name.

File: blockly-ws-search/workspace_search.js [hunk 2]
```diff
@@ -395,4 +401,75 @@ class WorkspaceSearch {
...
+
+    let targetLeft = metrics.viewLeft;
+
+    const overflowLeft = left < metrics.viewLeft;
+    const overflowRight = right > metrics.viewLeft + metrics.viewWidth;
+    const hasXOverflow = overflowLeft || overflowRight;
+    const wideBlock = width > metrics.viewWidth;
+
+    if ((overflowLeft && !wideBlock) ||
+        (wideBlock && !this.workspace_.RTL && hasXOverflow)) {
```
Comment (alschmiedt) [inline blockly-ws-search/workspace_search.js]: do we need the check for hasXOverflow here? It seems like if wide > metrics.viewWidth is true then there will always be an overflow?

**Review by alschmiedt — APPROVED** [review]
Looks great! Two small things then lgtm : )

File: blockly-ws-search/workspace_search.js (hunk 1 above)
Comment (moniika) [reply blockly-ws-search/workspace_search.js]: Cleaned it up. It looks nicer now.

File: blockly-ws-search/workspace_search.js (hunk 2 above)
Comment (moniika) [reply blockly-ws-search/workspace_search.js]: Yeah, not sure why I thought it was necessary. Removing it as I can't think of any case where it wouldn't be true.

File: blockly-ws-search/workspace_search.js [hunk 3]
```diff
@@ -395,4 +401,62 @@ class WorkspaceSearch {
...
+
+    // Block bounds in pixels relative to the workspace origin (0,0 is centre).
+    const width = block.width * scale;
+    const height = block.height * scale;
+    const top = xy.y * scale;
+    const bottom = (xy.y + block.height) * scale;
+    // In RTL the block's position is the top right of the block, not top left.
+    const left = this.workspace_.RTL ? xy.x * scale - width: xy.x * scale;
+    const right = this.workspace_.RTL ? xy.x * scale : xy.x * scale +  width;
+
```
Comment (alschmiedt) [inline blockly-ws-search/workspace_search.js:428]: Looks great : ) On another PR I would maybe make it blockTop, blockBottom etc. just for readability sake

**Pull request merged**